FightSequence.py

```diff
@@ -10,46 +10,57 @@ def __init__(
         self.enemy = enemy
 
     def Fight(
-        self, playerWeapon: Item.Weapon, enemyWeapon: Item.Weapon
+        self,
+        playerWeapon: Item.Weapon,
+        enemyWeapon: Item.Weapon,
+        playerDefense: Item.Armor,
+        enemyDefense: Item.Armor,
     ) -> tuple[CharacterFactory.Character, CharacterFactory.Character]:
         """
         Simulates a fight between player and enemy
         :param playerAttack: The attack the player is using
         :param enemyAttack: The attack the enemy is using
         :return: The stat changes for the player and enemy
         """
-
+        print("Fighting")
         canPlayerAttack = self.CanAttack(self.player, playerWeapon)
         canEnemyAttack = self.CanAttack(self.enemy, enemyWeapon)
 
         if canPlayerAttack:
-            playerDamage = self.CalculateDamage(playerWeapon, self.player, self.enemy)
+            playerDamage = self.CalculateDamage(self.player, playerWeapon, enemyDefense)
             self.player.curStamina -= playerWeapon.staminaCost
-            self.enemy.health -= playerDamage
-            fightScreen.UpdateMeters(self.player, self.player.meters)
+            self.enemy.curHealth -= playerDamage
         if canEnemyAttack:
-            enemyDamage = self.CalculateDamage(enemyWeapon, self.enemy, self.player)
+            enemyDamage = self.CalculateDamage(self.enemy, enemyWeapon, playerDefense)
             self.enemy.curStamina -= enemyWeapon.staminaCost
-            self.player.health -= enemyDamage
-            fightScreen.UpdateMeters(self.enemy, self.enemy.meters)
+            self.player.curHealth -= enemyDamage
 
         return self.player, self.enemy
 
-    def CanAttack(
-        self, player: CharacterFactory.Character, attack: Item.Weapon
-    ) -> bool:
+    def CanAttack(self, player: Character, attack: Item.Weapon) -> bool:
         """
         Checks to see if character has enough stamina or magic to attack with.
         """
         return (
             player.curStamina - attack.staminaCost >= 0
-            and player.curMagic >= attack.magicCost
+            and player.curMagic >= attack.magicReq
         )
 
-    def CalculateDamage(playerAttack: Item.Weapon, targetDefense: Item.Armor):
+    def CalculateDamage(
+        self, player: Character, playerAttack: Item.Weapon, targetDefense: Item.Armor
+    ):
         """
         Calculates damage based on playerAttack vs the targets defensive item.
         """
-        attackDamage = playerAttack.damage - targetDefense.armorVal
-
-        return attackDamage
+        if playerAttack.magic:
+            attackDamage = (
+                playerAttack.damage + player.intelligence
+            ) - targetDefense.magicDefense
+        else:
+            attackDamage = (
+                playerAttack.damage + player.strength
+            ) - targetDefense.armorVal
```

Comment:
We're going to need a write-up of what these stats do, cause I am lost.

Reply:
(total damage) - total defense

Reply:
Ohhh, now that I know `magic` is a boolean this makes a bit more sense. So the item can either do physical damage or magical damage? Both the types of damage come out of the players health, but they may have high physical blocking armor (like shields) but low magical blocking armor (like spells)?

Reply:
Yeah lowkey kinda forgot until I went back and Looked lmao